src/sources/util/unix_stream.rs

```diff
@@ -34,15 +35,20 @@ use crate::{
 /// syslog source).
 pub fn build_unix_stream_source(
     listen_path: PathBuf,
+    socket_file_mode: Option<u32>,
     decoder: Decoder,
     handle_events: impl Fn(&mut [Event], Option<Bytes>) + Clone + Send + Sync + 'static,
     shutdown: ShutdownSignal,
     out: SourceSender,
-) -> Source {
-    Box::pin(async move {
-        let listener = UnixListener::bind(&listen_path).expect("Failed to bind to listener socket");
-        info!(message = "Listening.", path = ?listen_path, r#type = "unix");
+) -> crate::Result<Source> {
+    let listener = UnixListener::bind(&listen_path).expect("Failed to bind to listener socket");
+    info!(message = "Listening.", path = ?listen_path, r#type = "unix");
 
+    if let Err(e) = change_socket_permissions(&listen_path, socket_file_mode) {
+        return Err(e);
+    }
```

Comment:
Ditto, and I'm _really_ surprised `clippy` doesn't pick up on this.

Reply:
Indeed :thinking:

Reply:
Ah yeah I remember :smile: 
```
error: unused `std::result::Result` that must be used
  --> src/sources/util/unix_stream.rs:47:5
   |
47 |     change_socket_permissions(&listen_path, socket_file_mode);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D unused-must-use` implied by `-D warnings`
   = note: this `Result` may be an `Err` variant, which should be handled
```

Reply:
It's not the same result type so can't use the `?`

Reply:
I think you want `change_socket_permissions(&listen_path, socket_file_mode)?;` (notice the `?` at the end). Or did you try that?

Reply:
Is there a missing `?` after the call? The error quoted above is missing it, and so Rust complains that the `Result` must be used. Regarding result types, I think that's handled by the operator, which expands out to:
```rust
match <expression> {
    Ok(ok) => ok,
    Err(error) => return Err(error.into()),
}
```

Reply:
Yeah sorry my bad, long day :sleeping: you're right! Fixed :sweat_smile: